implementation/src/main/java/io/smallrye/config/ExpressionConfigSourceInterceptor.java

```diff
@@ -53,25 +50,39 @@ private ConfigValue getValue(final ConfigSourceInterceptorContext context, final
             return null;
         }
 
-        List<Problem> problems = new ArrayList<>();
+        ConfigValue.ConfigValueBuilder value = configValue.from();
         Expression expression = Expression.compile(escapeDollarIfExists(configValue.getValue()), LENIENT_SYNTAX, NO_TRIM,
-                NO_SMART_BRACES);
+                NO_SMART_BRACES, DOUBLE_COLON);
```

Comment:
@radcortez hey Roberto, recently I had a chance to test this PR together with Keycloak. I encountered an issue with `quarkus.locales/quarkus.default-locale` properties. This issue probably affects even some other properties, the locale related ones were just first that I spotted. 
As soon as I added the SNAPSHOT package (with the changes from this PR included) to the Keycloak project, I got an error while trying to build the KC distribution:
```
SRCFG00014: The config property quarkus.locales is required but it could not be found in any config source
````
This was quite weird signal, because afaik by default the locale related properties come from the default config source (if they're not defined somewhere else) set by Quarkus.
Upon further debugging I found out `quarkus.locales` is indeed null, probably because the `${user.language:en}-${user.country:}` expression is not propagated/expanded properly (see the attached screenshot).

I went a little bit further and narrowed the pool of possible culprits to the `DOUBLE_COLON` expression flag. If I remove it and build the package again, everything seems to work properly.

I assume that this expression flag is used primarily for parsing/evaluating expressions like `${aes-gcm-nopadding::encoded::x}` in order to get the encoded part. But somehow it seems to be in collision with the type of expressions mentioned above.

By the way, thanks for all the effort you and your team do around the project! Possibility to secure secrets will help us a lot.

![telegram-cloud-photo-size-4-5974175736824707476-y](https://user-images.githubusercontent.com/12138171/216638073-961053ea-04e8-4839-9014-9a606d9f0842.jpg)

Reply:
Hey @Pepo48,

Thanks for taking the time to try the PR out and finding issues. Highly appreciated :)

Let me have a look.

Reply:
@radcortez you're right, we take our dependencies from quarkus-bom (currently in version 2.13.6.Final), which contains `smallrye-common-expression` version 1.13.1. I overrode only the `smallrye-config` dependencies. 

The bump resolved it. Sorry for bothering, I could've checked the common project's recent issues first.
Thanks for your time!

Reply:
No worries, it wasn't that evident :)

Please let me know how the API works and if it solves your issues. We don't have secret handlers implementations yet, but that will come. You can probably try the `jasypt` one or implement your own.

I'm happy to add any missing features or APIs.

Reply:
@Pepo48 wondering if you have any feedback to share? :)

Reply:
As I stated above, this one was solved by overriding the dependency as you suggested. Since then haven't encountered any other issues connected to it.